utils/src/main/scala/za/co/absa/enceladus/utils/implicits/StringImplicits.scala

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2018-2019 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.utils.implicits
+
+import java.security.InvalidParameterException
+
+import scala.annotation.tailrec
+
+object StringImplicits {
+  implicit class StringImprovements(string: String) {
+    def findFirstUnquoted(charsToFind: Set[Char], quoteChars: Set[Char], escape: Char = '\\' ): Option[Integer] = {
+      @tailrec
+      def scan(sub: String, index: Integer, outQuote: Option[Char], escaped: Boolean = false):Option[Integer] = {
+        //escaped flag defaults to false, as every non-escape character clears it
+        val head: Option[Char] = sub.find(_ => true) //get the first character of the string if it exists
+        (head, escaped) match {
+          case (None, _) => None // end of string
+          case (`outQuote`, false) if !outQuote.contains(escape)  => scan(sub.tail, index + 1, None) // end of quote,
+                                                                                                     // unless escaped
+                                                                                                     // or equals escape
```

Comment:
Do you think it would make it easier to understand the logic? I found it this way it made it clearer which line the explanation belongs to. Also easier to compare the cases...
Although the one comment with same indentation separates the later cases from the earlier ones, again for easier understanding of the flow.